dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -176,6 +177,32 @@ public WorkflowDAO(SessionFactory factory) {
         return filteredWorkflows.size() == 1 ? Optional.of(filteredWorkflows.get(0)) : Optional.empty();
     }
 
+    /**
+     * Find if a path already exists in the BioWorkflow or Apptool table since we do not want duplicate names between them.
+     * If creating an apptool, check the workflow table and vice versa.
+     *
+     * @param path
+     * @param clazz
+     * @param <T>
+     * @return
+     */
+    public <T extends Workflow> void checkForDuplicateAcrossTables(String path, Class<T> clazz) {
+        final List<Workflow> workflows = findByPath(path, false);
+        final List<T> filteredWorkflows = workflows.stream()
+            .filter(workflow -> workflow.getClass().equals(clazz))
+            .map(clazz::cast)
```

Comment:
I don't think you need this; you don't do anything with the resulting entries, so no need to cast. If so, see my suggestion about using `anyMatch`.